From fc907f7293edc59425fcc91bb1b684a7b02114f9 Mon Sep 17 00:00:00 2001
From: Simon Bruder <simon@sbruder.de>
Date: Sat, 25 Apr 2026 01:55:40 +0200
Subject: [PATCH 1/5] Load assets from local directory

---
 standalone/src/assets.js | 96 +++-------------------------------------
 1 file changed, 6 insertions(+), 90 deletions(-)

diff --git a/standalone/src/assets.js b/standalone/src/assets.js
index 4359a13..bf30a61 100644
--- a/standalone/src/assets.js
+++ b/standalone/src/assets.js
@@ -1,72 +1,8 @@
 import { cors } from "@elysiajs/cors";
 import { Elysia } from "elysia";
-import { db } from "./db.js";
+import path from "node:path";
 
-if (
-  process.env.ENABLE_ASSETS_SERVER === "true" &&
-  (process.env.WIDGET_VERSION === "latest" || process.env.WASM_VERSION === "latest")
-) {
-  console.warn(
-    "📦 [asset server] using 'latest' version for assets is not recommended for production!\n   make sure to pin it to a set version using the WIDGET_VERSION and WASM_VERSION env variables.",
-  );
-}
-
-const updateCache = async () => {
-  if (process.env.ENABLE_ASSETS_SERVER !== "true") return;
-
-  let cacheConfig = {};
-  try {
-    cacheConfig = JSON.parse((await db.get("asset:cache-config")) || "{}");
-  } catch { }
-
-  const lastUpdate = cacheConfig.lastUpdate || 0;
-  const currentTime = Date.now();
-  const updateInterval = 1000 * 60 * 60 * 24; // 1 day
-  const intervalExceeded = currentTime - lastUpdate > updateInterval;
-
-  const WIDGET_VERSION = process.env.WIDGET_VERSION || "latest";
-  const WASM_VERSION = process.env.WASM_VERSION || "latest";
-
-  if (!cacheConfig.versions) cacheConfig.versions = {};
-  const versionsChanged =
-    cacheConfig.versions.widget !== WIDGET_VERSION || cacheConfig.versions.wasm !== WASM_VERSION;
-
-  if (!intervalExceeded && !versionsChanged) return;
-
-  const CACHE_HOST = process.env.CACHE_HOST || "https://cdn.jsdelivr.net";
-
-  try {
-    const [widgetSource, floatingSource, wasmSource, wasmLoaderSource] = await Promise.all([
-      fetch(`${CACHE_HOST}/npm/@cap.js/widget@${WIDGET_VERSION}`).then((r) => r.text()),
-      fetch(`${CACHE_HOST}/npm/@cap.js/widget@${WIDGET_VERSION}/cap-floating.min.js`).then((r) =>
-        r.text(),
-      ),
-      fetch(`${CACHE_HOST}/npm/@cap.js/wasm@${WASM_VERSION}/browser/cap_wasm_bg.wasm`).then((r) =>
-        r.arrayBuffer(),
-      ),
-      fetch(`${CACHE_HOST}/npm/@cap.js/wasm@${WASM_VERSION}/browser/cap_wasm.min.js`).then((r) =>
-        r.text(),
-      ),
-    ]);
-
-    cacheConfig.lastUpdate = currentTime;
-    cacheConfig.versions.widget = WIDGET_VERSION;
-    cacheConfig.versions.wasm = WASM_VERSION;
-
-    await Promise.all([
-      db.set("asset:cache-config", JSON.stringify(cacheConfig)),
-      db.set("asset:widget.js", widgetSource),
-      db.set("asset:floating.js", floatingSource),
-      db.set("asset:cap_wasm_bg.wasm", Buffer.from(wasmSource)),
-      db.set("asset:cap_wasm.js", wasmLoaderSource),
-    ]);
-  } catch (e) {
-    console.error("📦 [asset server] failed to update assets cache:", e);
-  }
-};
-
-updateCache();
-setInterval(updateCache, 1000 * 60 * 60);
+const ASSETS_PATH = path.resolve("./assets");
 
 export const assetsServer = new Elysia({
   prefix: "/assets",
@@ -83,37 +19,17 @@ export const assetsServer = new Elysia({
   })
   .get("/widget.js", async ({ set }) => {
     set.headers["Content-Type"] = "text/javascript";
-    const content = await db.get("asset:widget.js");
-    if (!content) {
-      set.status = 503;
-      return "Asset not cached yet";
-    }
-    return content;
+    return Bun.file(path.resolve(ASSETS_PATH, "widget.js"));
   })
   .get("/floating.js", async ({ set }) => {
     set.headers["Content-Type"] = "text/javascript";
-    const content = await db.get("asset:floating.js");
-    if (!content) {
-      set.status = 503;
-      return "Asset not cached yet";
-    }
-    return content;
+    return Bun.file(path.resolve(ASSETS_PATH, "floating.js"));
   })
   .get("/cap_wasm_bg.wasm", async ({ set }) => {
     set.headers["Content-Type"] = "application/wasm";
-    const content = await db.getBuffer("asset:cap_wasm_bg.wasm");
-    if (!content) {
-      set.status = 503;
-      return "Asset not cached yet";
-    }
-    return content;
+    return Bun.file(path.resolve(ASSETS_PATH, "cap_wasm_bg.wasm"));
   })
   .get("/cap_wasm.js", async ({ set }) => {
     set.headers["Content-Type"] = "text/javascript";
-    const content = await db.get("asset:cap_wasm.js");
-    if (!content) {
-      set.status = 503;
-      return "Asset not cached yet";
-    }
-    return content;
+    return Bun.file(path.resolve(ASSETS_PATH, "cap_wasm.js"));
   });
-- 
2.51.2

